spec/models/trait_descriptor_spec.rb

```diff
@@ -2,129 +2,39 @@
 
 RSpec.describe TraitDescriptor do
   context "validations" do
-    it { should validate_presence_of(:descriptor_name) }
-    it { should validate_presence_of(:category) }
+    it { should validate_presence_of(:trait_id) }
+    it { should validate_presence_of(:scoring_method) }
     it { should validate_presence_of(:units_of_measurements) }
   end
 
   describe '#table_data' do
     it 'properly calculates associated trait score number' do
-      pts = create_list(:plant_trial, 2)
-      psu0 = create(:plant_scoring_unit, plant_trial: pts[0])
-      psu1 = create(:plant_scoring_unit, plant_trial: pts[1])
-      tses0 = create_list(:trait_score, 3, plant_scoring_unit: psu0, trait_descriptor: nil)
-      tses1 = create_list(:trait_score, 2, plant_scoring_unit: psu1, trait_descriptor: nil)
-      td1 = create(:trait_descriptor, trait_scores: [tses0[0], tses1[0], tses1[1]])
-      td2 = create(:trait_descriptor, trait_scores: [tses0[1], tses0[2]])
+      td1 = create(:trait_descriptor)
+      td2 = create(:trait_descriptor)
+      create_list(:trait_score, 3, trait_descriptor: td1)
+      create_list(:trait_score, 2, trait_descriptor: td2)
       table_data = TraitDescriptor.table_data
-      expect(table_data.count).to eq 3
-      expect(table_data.map{ |td| [td[2], td[3], td[5]] }).to match_array [
-        [td1.descriptor_name, pts[0].project_descriptor, '1'],
-        [td1.descriptor_name, pts[1].project_descriptor, '2'],
-        [td2.descriptor_name, pts[0].project_descriptor, '2']
+      expect(table_data.count).to eq 2
+      expect(table_data.map{ |td| [td[1], td[6]] }).to match_array [
+        [td1.trait.name, 3],
```

Comment:
Minor thingie - since `TraitDescriptor#trait_name` is defined, I think it should be preferred here.